flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -211,6 +220,58 @@ protected static Notification createNotification(
             .setOngoing(BooleanUtils.getValue(notificationDetails.ongoing))
             .setOnlyAlertOnce(BooleanUtils.getValue(notificationDetails.onlyAlertOnce));
 
+    if (notificationDetails.actions != null) {
+      // Space out request codes by 16 so even with 16 actions they won't clash
+      int requestCode = notificationDetails.id * 16;
```

Comment:
Is multiplying by 16 required? From what I could tell from the issue that had been reported before, the problem was to do with using the same request code (999) for all of the pending intents associated with an action. From what I read [here](https://stackoverflow.com/a/38759473) that also references the official docs, so long as the combination of receiver and request code are unique then there shouldn't be clashes. If users of the plugin provide unique action IDs, this should mean that multiplying by 16 is redundant 

cc @Kavantix

Reply:
Ah I see what you mean, I'm not 100% sure I tried to use the notification id directly, if that would also solve the problem then the times 16 should indeed not be needed

Reply:
Oops apologies I misread and thought it was a numeric action id being passed. In that case a solution that I reckon would be better is to remove this logic and then update the line further below so instead of

```
 PendingIntent.getBroadcast(
                context, requestCode++, actionIntent, PendingIntent.FLAG_ONE_SHOT)
```

it becomes

```
```
 PendingIntent.getBroadcast(
                context, action.id.hashCode(), actionIntent, PendingIntent.FLAG_ONE_SHOT)
```

to allow for using the hash code of the action id, which should be unique given action id is expected to be a unique identifier. Thoughts?

Reply:
Hashcodes are not guaranteed to be unique, even worse, this will make the request code of two different notifications the same since the action id will be the same and thus the hashcode

Reply:
Ah good point, been a while since I looked at them so was going by how a lot of equality checks I've seen of late includes it as part of the calculation. Regarding the second point, appending the notification id to the string could help but it's a moot point if the hash codes aren't unique